"panic: assignment to entry in nil map" on creation of azurerm_linux_web_app

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.2

### AzureRM Provider Version

3.63.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "lwa_cc" {
  name                = local.names.linux_web_app
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location
  service_plan_id     = azurerm_service_plan.sp.id

  site_config {
    application_stack {
      docker_image_name        = var.container_registry.docker_image
      docker_registry_url      = "https://${data.azurerm_container_registry.cr.login_server}"
      docker_registry_username = data.azurerm_container_registry.cr.admin_username
      docker_registry_password = data.azurerm_container_registry.cr.admin_password
    }
  }

  tags = local.default_tags
}
```


### Debug Output/Panic Output

```shell
azurerm_linux_web_app.lwa_cc: Creating...
╷
│ Error: Request cancelled
│
│   with azurerm_linux_web_app.lwa_cc,
│   on main.tf line 42, in resource "azurerm_linux_web_app" "lwa_cc":
│   42: resource "azurerm_linux_web_app" "lwa_cc" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.63.0_x5 plugin:

panic: assignment to entry in nil map

goroutine 125 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers.(*SiteConfigLinux).ExpandForCreate(0xc00297cb08, 0x0)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers/linux_web_app_schema.go:804 +0xe65
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.LinuxWebAppResource.Create.func1({0x834aa48, 0xc002788330}, {0xc002ba0000, {0x834b2d0, 0xc0003f7860}, 0xc0022cd600, 0x0, {0x834c450, 0xca45ed0}})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/linux_web_app_resource.go:325 +0xe10
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x834aa48, 0xc002788330}, 0x834aa80?, {0x6f1bf80?, 0xc002ba0000?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:55 +0x163
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x834aa48, 0xc002788330}, 0x0?, {0x6f1bf80, 0xc002ba0000})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:188 +0xb8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0000fb500, {0x834aa80, 0xc001ebb5f0}, 0xd?, {0x6f1bf80, 0xc002ba0000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000fb500, {0x834aa80, 0xc001ebb5f0}, 0xc0026c4b60, 0xc0022cd480, {0x6f1bf80, 0xc002ba0000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001ad0e40, {0x834aa80?, 0xc001ebb4d0?}, 0xc0026be0a0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0006c9b80, {0x834aa80?, 0xc001ebaa80?}, 0xc000746fc0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7719e00?, 0xc0006c9b80}, {0x834aa80, 0xc001ebaa80}, 0xc000746f50, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001d41e0, {0x835afe0, 0xc00071c680}, 0xc0026ae360, 0xc00184f7a0, 0xca06720, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc0001d41e0, {0x835afe0, 0xc00071c680}, 0xc0026ae360, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.63.0_x5 plugin crashed!
```


### Expected Behaviour

The azurerm_linux_web_app created with the provided configuration

```
Terraform will perform the following actions:
  # azurerm_linux_web_app.lwa_cc will be created
  + resource "azurerm_linux_web_app" "lwa_cc" {
      + client_affinity_enabled           = false
      + client_certificate_enabled        = false
      + client_certificate_mode           = "Required"
      + custom_domain_verification_id     = (sensitive value)
      + default_hostname                  = (known after apply)
      + enabled                           = true
      + hosting_environment_id            = (known after apply)
      + https_only                        = false
      + id                                = (known after apply)
      + key_vault_reference_identity_id   = (known after apply)
      + kind                              = (known after apply)
      + location                          = "westeurope"
      + name                              = "control-center-development"
      + outbound_ip_address_list          = (known after apply)
      + outbound_ip_addresses             = (known after apply)
      + possible_outbound_ip_address_list = (known after apply)
      + possible_outbound_ip_addresses    = (known after apply)
      + resource_group_name               = "Control-Center"
      + service_plan_id                   = "/subscriptions/***/resourceGroups/Control-Center/providers/Microsoft.Web/serverfarms/ASP-MyCC-tf"
      + site_credential                   = (sensitive value)
      + tags                              = {
          + "project" = "MyCC"
          + "source"  = "terraform"
        }
      + zip_deploy_file                   = (known after apply)

      + site_config {
          + always_on                               = true
          + container_registry_use_managed_identity = false
          + default_documents                       = (known after apply)
          + detailed_error_logging_enabled          = (known after apply)
          + ftps_state                              = "Disabled"
          + health_check_eviction_time_in_min       = (known after apply)
          + http2_enabled                           = false
          + linux_fx_version                        = (known after apply)
          + load_balancing_mode                     = "LeastRequests"
          + local_mysql_enabled                     = false
          + managed_pipeline_mode                   = "Integrated"
          + minimum_tls_version                     = "1.2"
          + remote_debugging_enabled                = false
          + remote_debugging_version                = (known after apply)
          + scm_minimum_tls_version                 = "1.2"
          + scm_type                                = (known after apply)
          + scm_use_main_ip_restriction             = false
          + use_32_bit_worker                       = true
          + vnet_route_all_enabled                  = false
          + websockets_enabled                      = false
          + worker_count                            = (known after apply)

          + application_stack {
              + docker_image_name        = "control-center:latest"
              + docker_registry_password = (sensitive value)
              + docker_registry_url      = "https://mydev01.azurecr.io"
              + docker_registry_username = "mydev01"
            }
        }
    }

```

### Actual Behaviour

╷
│ Error: Request cancelled
│
│   with azurerm_linux_web_app.lwa_cc,
│   on main.tf line 42, in resource "azurerm_linux_web_app" "lwa_cc":
│   42: resource "azurerm_linux_web_app" "lwa_cc" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/helpers/linux_web_app_schema.go
+++ b/internal/services/appservice/helpers/linux_web_app_schema.go
@@ -801,6 +801,9 @@ func (s *SiteConfigLinux) ExpandForCreate(appSettings map[string]string) (*web.S
 
 		if linuxAppStack.DockerImageName != "" {
 			expanded.LinuxFxVersion = pointer.To(EncodeDockerFxString(linuxAppStack.DockerImageName, linuxAppStack.DockerRegistryUrl))
+			if appSettings == nil {
+				appSettings = map[string]string{}
+			}
 			appSettings["DOCKER_REGISTRY_SERVER_URL"] = linuxAppStack.DockerRegistryUrl
 			appSettings["DOCKER_REGISTRY_SERVER_USERNAME"] = linuxAppStack.DockerRegistryUsername
 			appSettings["DOCKER_REGISTRY_SERVER_PASSWORD"] = linuxAppStack.DockerRegistryPassword

